src/gui/entry/EntryModel.cpp

```diff
@@ -145,17 +206,139 @@ QVariant EntryModel::data(const QModelIndex& index, int role) const
             }
             return result;
         case Username:
-            result = entry->resolveMultiplePlaceholders(entry->username());
+            /*
+             * Display usernames hidden or visible according to current state
+             * of 'Hide Usernames' setting
+             */
+            if (m_hideUsernames) {
+                result = EntryModel::HiddenContent;
+            }
+            else {
+                //result = entry->username();
+                result = entry->resolveMultiplePlaceholders(entry->username());
+            }
             if (attr->isReference(EntryAttributes::UserNameKey)) {
                 result.prepend(tr("Ref: ","Reference abbreviation"));
             }
             return result;
+        case Password:
+            /*
+             * Display passwords hidden or visible according to current state
+             * of 'Hide Passwords' setting
+             */
+            if (m_hidePasswords) {
+                result = EntryModel::HiddenContent;
+            }
+            else {
+                //result = entry->resolveMultiplePlaceholders(entry->password());
+                result = entry->password();
+            }
+            if (attr->isReference(EntryAttributes::PasswordKey)) {
+                result.prepend(tr("Ref: ","Reference abbreviation"));
+            }
+            return result;
         case Url:
+            //result = entry->resolveMultiplePlaceholders(entry->displayUrl());
             result = entry->displayUrl();
             if (attr->isReference(EntryAttributes::URLKey)) {
                 result.prepend(tr("Ref: ","Reference abbreviation"));
             }
             return result;
+        case Notes:
```

Comment:
I don't see the value in this, they are now displayed in the details pane below the entry list. Similar to attachments, it might just be ok to write "YES"/"NO".

Reply:
After thinking about it, and the fact that you truncate the length, I am OK with this approach. Please disregard this comment.

Reply:
ok, thanks